File: src/main/java/kr/co/lunatalk/domain/product/repository/ProductRepositoryCustom.java
package kr.co.lunatalk.domain.product.repository;

import kr.co.lunatalk.domain.product.domain.Product;
import kr.co.lunatalk.domain.product.dto.FindProductDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface ProductRepositoryCustom {
	Optional<Product> findProductById(Long productId);
	List<Product> findAllProductsByProductIds(List<Long> productIds);
	List<Product> findAllProductDtoByIdsWithJoin(List<Long> productIds);
	Page<Product> findAll(String productName, Pageable pageable);
}

File: src/main/kotlin/kr/co/lunatalk/domain/product/repository/ProductRepositoryCustom.kt
package kr.co.lunatalk.domain.product.repository

import kr.co.lunatalk.domain.product.domain.Product
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.Optional

interface ProductRepositoryCustom {
    fun findProductById(productId: Long): Optional<Product>
    fun findAllProductsByProductIds(productIds: List<Long>): List<Product>
    fun findAllProductDtoByIdsWithJoin(productIds: List<Long>): List<Product>
    fun findAll(productName: String?, pageable: Pageable): Page<Product>
}
